src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -489,14 +489,13 @@ class SelectTargetElementSegment(BaseSegment):
     # Important to split elements before parsing, otherwise debugging is really hard.
     match_grammar = GreedyUntil(Ref('CommaSegment'))
     parse_grammar = OneOf(
-        # *
-        Ref('StarSegment'),
-        # blah.* + blah.blah.*
+        # *, blah.*, blah.blah.*, etc.
         Sequence(
-            Ref('SingleIdentifierGrammar'), Ref('DotSegment'),
-            Sequence(
-                Ref('SingleIdentifierGrammar'), Ref('DotSegment'),
-                optional=True
+            AnyNumberOf(
+                Sequence(
+                    Ref('SingleIdentifierGrammar'),
+                    Ref('DotSegment')
```

Comment:
@Dandandan - can you also add a `code_only=True` here, I think it will speed up parsing slightly because it will bypass the whitespace routines. Otherwise this looks great.

Reply:
Thanks, done!